packages/core/src/modules/oob/OutOfBandService.ts

```diff
@@ -28,6 +38,43 @@ export class OutOfBandService {
     this.eventEmitter = eventEmitter
   }
 
+  public async createImplicitInvitation(
+    agentContext: AgentContext,
+    config: CreateImplicitInvitationConfig
+  ): Promise<OutOfBandRecord> {
+    const { did, threadId, handshakeProtocols, autoAcceptConnection, recipientKey } = config
+
+    // Verify it is a valid did and it is present in the wallet
+    const publicDid = parseDid(did)
+    if (!(await agentContext.dependencyManager.resolve(DidsApi).getCreatedDids({ did: publicDid.did }))) {
+      throw new AriesFrameworkError(`Referenced public did ${did} not found.`)
+    }
+
+    const outOfBandInvitation = new OutOfBandInvitation({
+      id: did,
```

Comment:
Is the did the id so I can't make multiple?

Reply:
I guess this is answered in other comments. It has some drawbacks but I think it's a bit more clear than previous approach.